### PR TITLE
[3006.x] Refactor the `tools pkg repo` commands into a subdirectory

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
See title

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by MKLeb** [comment]
Passing staging run: https://github.com/saltstack/salt/actions/runs/4867253303

**Review by s0undt3ch — DISMISSED** [review]

File: tools/pkg/repo/__init__.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
+"""
+These commands are used to build the pacakge repository files.
```
Comment (twangboy) [inline tools/pkg/repo/__init__.py:2]: package

File: tools/pkg/repo/create.py [hunk 2]
```diff
@@ -0,0 +1,1038 @@
+"""
+These commands are used to build the pacakge repository files.
```
Comment (twangboy) [inline tools/pkg/repo/create.py:2]: package

File: tools/pkg/repo/publish.py [hunk 3]
```diff
@@ -0,0 +1,653 @@
+"""
+These commands are used to build the pacakge repository files.
```
Comment (twangboy) [inline tools/pkg/repo/publish.py:2]: package

File: tools/pkg/repo/publish.py [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+)
+
+
+@publish.command(
+    arguments={
+        "repo_path": {
+            "help": "Local path for the repository that shall be published.",
+        },
+        "salt_version": {
+            "help": "The salt version for which to build the repository",
```
Comment (twangboy) [inline tools/pkg/repo/publish.py:58]: Is it building the repository here? Or publishing? This looks like a copy/paste from the `create.py`

File: tools/pkg/repo/publish.py [hunk 5]
```diff
@@ -0,0 +1,653 @@
...
+    )
+
+
+@publish.command(
+    arguments={
+        "repo_path": {
+            "help": "Local path for the repository that shall be published.",
+        },
+        "salt_version": {
+            "help": "The salt version for which to build the repository",
```
Comment (twangboy) [inline tools/pkg/repo/publish.py:80]: same here

**Review by twangboy — CHANGES_REQUESTED** [review]

**Pull request merged**